### ISSUE TITLE
Removal of js address form validation in 1.7.5.2

### ISSUE THREAD

**Issue by Daaaaad** [PrestaShop/PrestaShop#14382]
Why do you removed the JS validation (that exists on PS 1.6) while editing an address in front?

Sometimes it's preferable to give the user an instant feedback 🤔

**Comment by khouloudbelguith** [comment]
Hi @Daaaaad,

Thanks for your report.
With PS1.6, before clicking on the "Save" button, we have js validation.
In the PS1.7, we need to click on save, to see the error
![image](https://user-images.githubusercontent.com/16067358/60185446-db2ccd80-9821-11e9-8d72-94ce668ed057.png)

ping @marionf, @colinegin what do you think?

Thanks!

**Comment by ghost** [comment]
https://bootstrap-validate.js.org/usage.html

**Comment by Daaaaad** [comment]
Yes @khouloudbelguith, that's what I noticed 😉

I would like to validate the VAT number using Javascript code (https://www.braemoor.co.uk/software/vat.shtml)

Before, there was a built-in process (js/validate.js) that we could use in combination with the "data-validation" attribute to validate the field and show instant feedback to the customer.

Furthermore, you removed the functionality but forgot to remove validate.js file that is no more used (I think).

In version 1.7+, am I forced to use server-side validation?

Thank you for the link @PrestaShark, but that bothers me to re-code something that was working (well) in previous versions 😔

Regards,

David.

**Comment by khouloudbelguith** [comment]
@Daaaaad, no it is used.
You can check this PR: https://github.com/PrestaShop/PrestaShop/pull/13604/files#diff-a9ae2f1b0bccf85788172a54b0eaa56cR102 as an example.

Thanks!

**Comment by Daaaaad** [comment]
Maybe I'm wrong @khouloudbelguith but I don't think _validate_isCustomerName_ is used... 

I just tested on version 1.7.5.2 and the function don't seems to be called 🤔

**Comment by khouloudbelguith** [comment]
@Daaaaad, it is used also in PS1.7.5.2, here's a PR: https://github.com/PrestaShop/PrestaShop/pull/13599/files#diff-a9ae2f1b0bccf85788172a54b0eaa56cR102 milestoned to PS1.7.5.2.
Thanks!

**Comment by Daaaaad** [comment]
Sorry again @khouloudbelguith but I really can't find where those functions (validate_isCustomerName/validate_isName) are called. Validate.js file don't seems to be included either 😕

**Comment by khouloudbelguith** [comment]
ping @PierreRambaud, @PrestaShop/prestashop-core-developers what do you think?

Thanks!

**Comment by PierreRambaud** [comment]
These functions still exist in PrestaShop 1.7 but are not used anymore in the classic theme.
And as I know, you're still able to add in your input
`data-validate="isCustomerName"` which will automatically run the `validate_isCustomerName` or 
`data-validate="isName"` which will automatically run the `validate_isName`

**Comment by Daaaaad** [comment]
You are right @PierreRambaud, I still can use `data-validate="isCustomerName"`... but I will have to edit the theme's templates whereas before the field definition was enough. And as I'm trying to create a module, that's a huge difference 😉

**Comment by colinegin** [comment]
@PierreRambaud what do you think of this suggestion ?

**Comment by PierreRambaud** [comment]
I don't know why they removed this behavior in 1.7, but we can restore it in the 1.7.7, it will not take long

**Comment by Hlavtox** [comment]
No longer relevant
